Composer/packages/client/src/components/ProjectTree/treeItem.tsx

```diff
@@ -201,18 +209,46 @@ const itemName = (nameWidth: number) => css`
 const calloutRootStyle = css`
   padding: 11px;
 `;
+
+const overflowButtonContainer = css`
+  display: flex;
+`;
+
+const actionButtonStyle = (isActive: boolean) => css`
+  display: flex;
+  justify-content: center;
+  height: 24px;
+  width: 24px;
+  .action-btn {
+    visibility: ${isActive ? 'visible' : 'hidden'};
+  }
+  :hover {
+    background: ${NeutralColors.gray20};
+    .action-btn {
+      visibility: visible;
+      path {
+        fill: #0078d4;
+      }
+    }
+  }
+`;
+
 // -------------------- TreeItem -------------------- //
 
 interface ITreeItemProps {
   link: TreeLink;
+  ActionIcon?: React.FC<any>;
```

Comment:
This is awkward way of doing this. Instead pass an onRenderActionIcon instead: (whatever args you want) => React.ReactNode